core/src/test/java/org/elasticsearch/action/admin/cluster/node/tasks/TasksIT.java

```diff
@@ -470,6 +480,76 @@ public void testTasksWaitForAllTask() throws Exception {
         assertThat(response.getTasks().size(), greaterThanOrEqualTo(1));
     }
 
+    public void testTaskResultPersistence() throws Exception {
+        // First make sure that template was correctly created
+        GetIndexTemplatesResponse response =
+            client().admin().indices().prepareGetTemplates(TaskResultsService.TASK_RESULT_INDEX_TEMPLATE).get();
+        assertEquals(1, response.getIndexTemplates().size());
+
+        registerTaskManageListeners(TestTaskPlugin.TestTaskAction.NAME);  // we need this to get task id of the process
+
+        // Start non-blocking test task
+        TestTaskPlugin.TestTaskAction.INSTANCE.newRequestBuilder(client()).setShouldPersistResult(true).setShouldBlock(false).get();
```

Comment:
I am not sure I see the point. If you get ActionResponse - you already have the result. Why would you ever want to re-read _exactly_ the same result from the index?